File: src/types/transforms.ts
export type TransformType =
  | "bin"
  | "recode"
  | "date_part"
  | "log"
  | "sqrt"
  | "zscore"
  | "arithmetic";

export interface TransformRequest {
  transform_type: TransformType;
  output_column: string;
  source_column: string;
  params: Record<string, unknown>;
}

export interface TransformResponse {
  output_column: string;
  row_count: number;
  null_count: number;
  dtype: string;
  preview: Array<string | number | boolean | null>;
}

export interface TransformTypeOption {
  type: TransformType;
  label: string;
}

export interface TransformListResponse {
  transforms: TransformTypeOption[];
}

export interface DeleteTransformResponse {
  ok: boolean;
  column_name: string;
}

File: backend/app/models/transforms.py
"""Pydantic models for column transform operations."""

from typing import Any
from pydantic import BaseModel, Field


class TransformRequest(BaseModel):
    """Request to create a computed column."""

    transform_type: str
    output_column: str
    source_column: str
    params: dict[str, Any] = Field(default_factory=dict)


class TransformResponse(BaseModel):
    """Response after creating a computed column."""

    output_column: str
    row_count: int
    null_count: int
    dtype: str
    preview: list


class TransformListResponse(BaseModel):
    """Available transform types."""

    transforms: list[dict[str, str]]
